src/sentry_slice.c

```diff
@@ -13,3 +23,73 @@ sentry__slice_eq(sentry_slice_t a, sentry_slice_t b)
 {
     return a.len == b.len && memcmp(a.ptr, b.ptr, a.len) == 0;
 }
+
+sentry_slice_t
+sentry__slice_split_at(sentry_slice_t a, char c)
+{
+    for (size_t i = 0; i < a.len; i++) {
+        if (a.ptr[i] == c) {
+            a.len = i;
+            return a;
+        }
+    }
+    return a;
+}
+
+size_t
+sentry__slice_find(sentry_slice_t a, char c)
+{
+    for (size_t i = 0; i < a.len; i++) {
+        if (a.ptr[i] == c) {
+            return i;
+        }
+    }
+    return (size_t)-1;
```

Comment:
I think its a good idea to have a sentinel value here (think `Option<usize>`), otherwise its too easy to go OOB.

Reply:
How could you go out of bounds? You can just unconditionally use the value. Whenever you subslice, you'd essentially create a slice with `len=0`, and at least then you never have to check for the sentinel value. And to check, you just compare with `len` instead of `-1`.

Reply:
https://github.com/getsentry/sentry-native/pull/178/files#diff-8c43795e4ec549677d4c8ab337de6a87R63
I think this will be a problem if `next == len`… `find` should always return a *valid* index, which `len` isn’t.